tree/splitter.h

```diff
@@ -0,0 +1,50 @@
+#ifndef SPLITTER_H__
+#define SPLITTER_H__
+
+/**
+ * Information regarding the split of a given node.
+ * This will be used by the tree during its construction.
+ */
+struct SplitterRecord
+{
+  double threshold;       /**<  Threshold to split at. */
+  double improvement;     /**<  Impurity improvement given parent node. */ 
+  double impurity_left;   /**<  Impurity of the left split. */
+  double impurity_right;  /**<  Impurity of the right split. */
+};
+
+/**
+ * Base class for all Splitters. The Splitter will be used by the tree to find
+ *  the best split point considering any given criterion and the Splitter strategy.
+ */
+class Splitter{
+  public:
+    virtual ~Splitter() { }
```

Comment:
The compiler automatically generates a destructor like this, we can safely omit it.